fix exceptions when processing conference papers

drh-stanford opened the pull request:
This PR fixes #77. I've tested this on a variety of conference and non-conference papers. It's ready for review.

darrenleeweber commented on spec/models/pub_hash_spec.rb:
```diff
@@ -809,6 +809,31 @@ def expect_field_match(field)
+      end
+      it 'creates citation data for event-place' do
+        expect(pub_hash.csl_doc).to include('event-place' => 'California')
+      end
```
Add a spec for a case with a city and no state/country.

darrenleeweber commented on app/models/pub_hash.rb:
```diff
@@ -148,8 +148,8 @@ def csl_doc
-          cit_data_hash['event-place'] = pub_hash[:conference][:city]
-          cit_data_hash['event-place'] << ',' unless pub_hash[:conference][:city].blank? || pub_hash[:conference][:statecountry].blank?
+          cit_data_hash['event-place'] = pub_hash[:conference][:city] || ''
+          cit_data_hash['event-place'] << ',' unless cit_data_hash['event-place'].blank?
```
This might result in an extra `,` after the city when we have a city but no state/country.

darrenleeweber commented:
Do we have time to add specs for `script/rebuild_conference_papers.rb` to check that it corrects the required pubs correctly?

drh-stanford commented:
I've corrected the problem when there a city but no state and added specs.

Re: script specs. I'm not sure how we can add specs since the scriptie is destructive.  Strangely, there's no specs for `Publication.rebuild_pub_hash`, so maybe the action item is to add some there?
